### PR TITLE
[scoped] include typo3 rector in scoped version follow up

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

**Comment by sabbelasichon** [comment]
Yeah. I will gonna have a closer look in the morning.

**Comment by sabbelasichon** [comment]
@TomasVotruba Do you need access to my repository to run the tests? Whom should i give access? You as a person are already having access.

**Comment by TomasVotruba** [comment]
@sabbelasichon Should be public repository for everyone. I see this outout:

```bash
  Retrieving the default branch name
  Not Found
```

**Comment by TomasVotruba** [comment]
Ah, it seem the repository does exist: https://github.com/ssch/typo3-rector/

image

It's the composer name vs repository name issue :)

File: .github/workflows/packages_tests.yaml [hunk 1]
```diff
@@ -27,21 +27,22 @@ jobs:
                     - rectorphp/rector-phpunit
                     - rectorphp/rector-cakephp
                     - rectorphp/rector-doctrine
+                    - ssch/typo3-rector
```
Comment (sabbelasichon) [inline .github/workflows/packages_tests.yaml]: this must be sabbelasichon/typo3-rector 
Sorry but this is history.

**Comment by sabbelasichon** [comment]
There is still one caveat regarding the scoping: https://github.com/sabbelasichon/typo3-rector/blob/main/src/Helper/StringUtility.php#L15

This changed to:
https://github.com/sabbelasichon/typo3-rector-prefixed/commit/11b2d0f7ca92b0270d628d48266be02786d239e1#diff-7de568990ddb061d79da36c08766c5112b233861212999bb3f66abf15a20eff7L12

Quick fix for me: https://github.com/sabbelasichon/typo3-rector/blob/v0.10.6/scoper.php#L37

**Comment by sabbelasichon** [comment]
Here you can see the result in rector itself:
https://github.com/rectorphp/rector/blob/main/vendor/ssch/typo3-rector/src/Helper/StringUtility.php#L12

**Comment by TomasVotruba** [comment]
I see. Thanks.
That's a bug in php-scoper. Should be kept untouched. I'll try to unscope it instead, it's more reliable solution

**Comment by TomasVotruba** [comment]
Looking at it there is lot of changes going on, the 0 is also removed... quite hard to understand the unscope.

What does it suppose to do? 

We already use string package that is handling string operations, e.g.:

```php
$stringy = new Stringy($variableName);
        return (string) $stringy->underscored()
            ->toUpperCase();
```

It would help to use that to have less code with same purpose. The unscoping is really mind bogling in this caes :D

**Comment by sabbelasichon** [comment]
I will have look into it, to get rid of this magic here.

**Comment by sabbelasichon** [comment]
Released version 0.11.6 with Stringy. Problem should be fixed. Thanks for pointing me.

**Comment by TomasVotruba** [comment]
Thanks :+1: 

Let's try it out in the wild :)

**Comment by sabbelasichon** [comment]
Thank you so much. I tested it and some other people are going to test it. It works like a charm (as expected ;-))

**Comment by sabbelasichon** [comment]
@TomasVotruba Found a weird thing. All ObjectTypes are prefixed:

https://github.com/rectorphp/rector/blob/main/vendor/ssch/typo3-rector/src/Rector/v9/v4/UseContextApiRector.php#L51

In rector-nette not:
https://github.com/rectorphp/rector/blob/main/vendor/rector/rector-nette/src/Rector/FuncCall/JsonDecodeEncodeToNetteUtilsJsonDecodeEncodeRector.php#L104

Will try to figure it out.

**Comment by sabbelasichon** [comment]
Migrations should be excluded from scoping also: https://github.com/rectorphp/rector/tree/main/vendor/ssch/typo3-rector/Migrations

**Comment by sabbelasichon** [comment]
Found it. Will do another Pull Request.

**Comment by TomasVotruba** [comment]
Great!
I'm trying to fix fail on PHP 7.1 build - https://github.com/rectorphp/rector/runs/2619865486

https://github.com/rectorphp/rector-src/commit/84802f54d62a5718c18b9a660d6925513d0022f0

**Comment by sabbelasichon** [comment]
Maybe this failure is related to: https://github.com/rectorphp/rector-src/blob/main/src/Contract/Rector/PhpRectorInterface.php#L24

The interface definition is wrong or does not contain ?Node as return type.

**Comment by TomasVotruba** [comment]
It's correct, it can return one node, many nodes or null.

**Comment by TomasVotruba** [comment]
The return type is correct, it can be more precise than parent from 7.1 to 8.0: https://3v4l.org/ePPYj

It's the removed param type thas is the problem: https://3v4l.org/H0GkA
It seems like some of your rules is passing something else then `PhpParser\Node`

**Comment by TomasVotruba** [comment]
AFAIK the trait in Rector rule might be a potential cause too, but that's just a guess.
Traits and param contravariancy are hard to solve.

**Comment by sabbelasichon** [comment]
Got it.

**Comment by sabbelasichon** [comment]
https://github.com/sabbelasichon/typo3-rector/commit/ab33106ea7cd0baeb29647a794d628bd857e0ea1

**Pull request merged**